wpilibc/src/main/native/cpp/PowerDistribution.cpp

```diff
@@ -87,6 +94,17 @@ double PowerDistribution::GetCurrent(int channel) const {
   return current;
 }
 
+std::vector<double> PowerDistribution::GetAllCurrents() const {
```

Comment:
@PeterJohnson should we make this be a SmallVector too? This could be a pretty hot path.

Reply:
I choose std::vector as it was the return type of all the NetworkTables get__Array().  I'm neutral either way.

The API call here should be the vector/SmallVector though, as I don't see anywhere in the C++ APIs that we return arrays.

Honestly, if speed was a concern, I think in the JNI service, we could go back to my initial prototype of just a straight double[24] array.  We don't need a vector there as we could just allocate to the biggest PDP size, and the HAL filling the array means it doesn't need to be initialized too.  Since it doesn't leave the function, we don't need the dynamic sizing there.  I just didn't see as much array support in C++ (and I'm not a native C++ developer).

Reply:
So typically how we do this is we have both a vector and a SmallVector overload of the same function.  The SmallVector one would be `std::span<double> GetAllCurrents(wpi::SmallVectorImpl<double>& buf)`.